File: RoadSense-Automation/Selenium-NodeJS/pages/getStartedPage.js
const { By } = require("selenium-webdriver");
const BasePage = require("./basePage");

class GetStartedPage extends BasePage {
  constructor(driver) {
    super(driver);
    this.tabCreateAccount = By.xpath("//div[contains(text(),'Create Account')]");
    this.tabSignIn = By.xpath("//div[contains(text(),'Sign In')]");
    this.inputName = By.xpath("//input[@placeholder='Full Name']");
    this.inputEmail = By.xpath("//input[@placeholder='Email Address']");
    this.inputPassword = By.xpath("//input[@placeholder='Password']");
    this.inputPhone = By.xpath("//input[@placeholder='Phone Number']");
    this.btnSubmit = By.xpath("//button[contains(text(),'Create Account') or contains(text(),'Sign In')]");
    this.textErrorMessage = By.xpath("//div[contains(@class,'error') or contains(text(),'Invalid')]");
  }

  async register(name, email, password, phone = "+91 98765 43210") {
    await this.click(this.tabCreateAccount);
    await this.sendKeys(this.inputName, name);
    await this.sendKeys(this.inputEmail, email);
    await this.sendKeys(this.inputPassword, password);
    await this.sendKeys(this.inputPhone, phone);
    await this.click(this.btnSubmit);
  }

  async login(email, password) {
    await this.click(this.tabSignIn);
    await this.sendKeys(this.inputEmail, email);
    await this.sendKeys(this.inputPassword, password);
    await this.click(this.btnSubmit);
  }

  async getErrorMessage() {
    if (await this.isDisplayed(this.textErrorMessage)) {
      return await this.getText(this.textErrorMessage);
    }
    return "";
  }
}

module.exports = GetStartedPage;

File: RoadSense-Automation/Appium-Python/pages/get_started_page.py
from appium.webdriver.common.appiumby import AppiumBy
from pages.base_page import BasePage

class GetStartedPage(BasePage):
    TAB_CREATE_ACCOUNT = (AppiumBy.XPATH, "//android.widget.TextView[@text='Create Account']")
    TAB_SIGN_IN = (AppiumBy.XPATH, "//android.widget.TextView[@text='Sign In']")
    TAB_RESET_CODE = (AppiumBy.XPATH, "//android.widget.TextView[@text='Reset Code']")

    INPUT_NAME = (AppiumBy.XPATH, "//android.widget.EditText[@hint='Full Name' or contains(@text, 'Alex')]")
    INPUT_EMAIL = (AppiumBy.XPATH, "//android.widget.EditText[@hint='Email Address' or contains(@text, 'email')]")
    INPUT_PASSWORD = (AppiumBy.XPATH, "//android.widget.EditText[@hint='Password' or @password='true']")
    INPUT_PHONE = (AppiumBy.XPATH, "//android.widget.EditText[@hint='Phone Number' or contains(@text, '98765')]")
    INPUT_VEHICLE = (AppiumBy.XPATH, "//android.widget.EditText[@hint='Vehicle Model' or contains(@text, 'Model')]")

    BTN_SUBMIT_SIGNUP = (AppiumBy.XPATH, "//android.widget.Button[contains(@text, 'Create Account') or contains(@text, 'Register')]")
    BTN_SUBMIT_LOGIN = (AppiumBy.XPATH, "//android.widget.Button[contains(@text, 'Sign In') or contains(@text, 'Login')]")
    BTN_SUBMIT_RESET = (AppiumBy.XPATH, "//android.widget.Button[contains(@text, 'Reset')]")

    TEXT_ERROR_MESSAGE = (AppiumBy.XPATH, "//android.widget.TextView[contains(@text, 'Error') or contains(@text, 'failed') or contains(@text, 'required')]")

    def register_user(self, name: str, email: str, password: str, phone: str = "+91 98765 43210", vehicle: str = "Tesla Model 3"):
        self.click(self.TAB_CREATE_ACCOUNT)
        self.send_keys(self.INPUT_NAME, name)
        self.send_keys(self.INPUT_EMAIL, email)
        self.send_keys(self.INPUT_PASSWORD, password)
        self.send_keys(self.INPUT_PHONE, phone)
        self.click(self.BTN_SUBMIT_SIGNUP)

    def login_user(self, email: str, password: str):
        self.click(self.TAB_SIGN_IN)
        self.send_keys(self.INPUT_EMAIL, email)
        self.send_keys(self.INPUT_PASSWORD, password)
        self.click(self.BTN_SUBMIT_LOGIN)

    def get_error_message(self) -> str:
        if self.is_displayed(self.TEXT_ERROR_MESSAGE):
            return self.get_text(self.TEXT_ERROR_MESSAGE)
        return ""
